src/app/body/product/product.component.ts

```diff
@@ -1,25 +1,51 @@
-import { Component, OnInit } from '@angular/core';
-import { ActivatedRoute } from '@angular/router';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { Product } from '../../models/product.model';
-import {ProductService} from '../../services/product.service';
+import { ProductService } from '../../services/product.service';
+import { ActivatedRoute, Params } from '@angular/router';
+import { Subject } from 'rxjs';
 
 @Component({
   selector: 'app-product',
   templateUrl: './product.component.html',
   styleUrls: ['./product.component.scss']
 })
-export class ProductComponent implements OnInit {
+export class ProductComponent implements OnInit, OnDestroy {
+  private destroy$ = new Subject<undefined>();
   public product: Product;
   public productRating: string[];
+  public proposals: Product[];
 
-  constructor(private productService: ProductService) { }
+  constructor(private productService: ProductService,
+              private route: ActivatedRoute) {
+    this.route.params.pipe().subscribe((params: Params) => {
+      this.product = this.productService.product;
+    });
```

Comment:
this should be in different method, try to keep constructor clean.
Also, useless pipe(), and why you need params here?

Reply:
fixed